src/main/java/org/jboss/logmanager/MDC.java

```diff
@@ -16,16 +16,32 @@
 
 package org.jboss.logmanager;
 
+import java.util.Iterator;
 import java.util.Map;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
 
 /**
  * Mapped diagnostic context.  This is a thread-local map used to hold loggable information.
  */
 public final class MDC {
+    private static final MDCProvider mdcProvider = getMDCProvider();
```

Comment:
Is it always true that if this class is loaded it will actually be called sooner or later?
If not then something like https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom could be used to defer the ServiceLoader overhead. Just saying...

Reply:
I think there would be no other reason to ever refer to this class than to immediately use it: it is an API of static methods.

Reply:
Ok, thanks for your feedback!